### PR TITLE
adding profiles for Trimaker printers

### PR DESCRIPTION
**Pull request by tomasbaldi**
Modification includes COSMOS II and NEBULA from Trimaker.

### CODE REVIEW

**Comment by tomasbaldi** [comment]
why it generates an error?

**Comment by nallath** [comment]
> why it generates an error?

Because we made a mistake with our CI system. It's entirely our fault.

**Comment by tomasbaldi** [comment]
should i post it again? or wait? Thanks!

**Comment by nallath** [comment]
Just wait :) No worries.

**Comment by konskarm** [comment]
@tomasbaldi the tests ran properly now. Please have a look at the errors.

**Comment by tomasbaldi** [comment]
@Ghostkeeper thank you very much! I am new in github!

**Comment by tomasbaldi** [comment]
I don't understand why it keeps failing

File: resources/extruders/trimaker_nebula_extruder.def.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "version": 2,
+    "name": "Extruder 1",
+    "inherits": "fdmextruder",
+    "metadata": {
+        "machine": "Trimaker Nebula",
```
Comment (nallath) [inline resources/extruders/trimaker_nebula_extruder.def.json:6]:
```suggestion
        "machine": "trimaker_nebula",
```
It needs to reference to the ID of the machine, not the name.

File: resources/extruders/trimaker_cosmosII_extruder.def.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    "version": 2,
+    "name": "Extruder 1",
+    "inherits": "fdmextruder",
+    "metadata": {
+        "machine": "Trimaker Cosmos II",
```
Comment (nallath) [inline resources/extruders/trimaker_cosmosII_extruder.def.json:6]:
```suggestion
        "machine": "trimaker_cosmosII",
```

**Comment by Ghostkeeper** [comment]
A couple of issues are still left, as found by the automated tests:
* `trimaker_nebula_platform.stl` is 2MB. We limit the mesh files to 1MB to reduce download and storage size of Cura (there are ~300 printers in Cura, so this would grow pretty quickly otherwise).
* `trimaker_cosmosII.def.json` refers to a platform `trimaker_cosmosII_platform.stl` that doesn't exist.
* `trimaker_cosmosII.def.json` and `trimaker_nebula.def.json` override `default_value`s for `infill_pattern` and `xy_offset_layer_0` which are not applied, since those settings already have a `value` property. Try overriding `value` instead.

**Comment by konskarm** [comment]
@tomasbaldi Could you have a look into @Ghostkeeper's comment?

**Comment by tomasbaldi** [comment]
Sorry for the delay. Thank you very much for your contributions, they are very helpful to me. I just posted a new commit. I just need to raise the mesh again after lowering the quality.

**Comment by tomasbaldi** [comment]
Thank you very much everyone for the help!

**Comment by tomasbaldi** [comment]
Now we just have to wait?

**Comment by tomasbaldi** [comment]
something happened with the request?

**Comment by konskarm** [comment]
Hi @tomasbaldi. I was just looking into the PR. I notice three things:

* Both the buildplate platforms seem misaligned (although I am not entirely sure if Cosmos II is misaligned):
image
* I am not sure if the Trimaker printers will be sold outside of Argentina, but I would suggest to name the quality profiles using English words
* The size of Trimerk Nebula is set to 220x220x260 in Cura, while the [Trimaker Nebula page](https://trimaker.com/nebula/) says it is 230x230x260. Which one is the correct one?

File: resources/definitions/trimaker_nebula.def.json [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        "machine_height": {"default_value": 260},
+
+        "layer_height": {"default_value": 0.2},
+        "material_flow": {"value": 100},
+        "xy_offset": {"default_value": 0},
+        "xy_offset_layer_0": {"value": -0.1},
+        "wall_thickness": {"value": "line_width * 3" },
+        "top_bottom_thickness": {"value":  "layer_height * 6"},
+        "infill_sparse_density": {"default_value": 25},
+        "infill_pattern": {"value": "grid"},
```
Comment (konskarm) [inline resources/definitions/trimaker_nebula.def.json:169]:
```suggestion
        "infill_pattern": {"value": "'grid'"},
```

File: resources/definitions/trimaker_cosmosII.def.json [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        "machine_height": {"default_value": 200},
+
+        "layer_height": {"default_value": 0.2},
+        "material_flow": {"value": 100},
+        "xy_offset": {"default_value": 0},
+        "xy_offset_layer_0": {"value": -0.1},
+        "wall_thickness": {"value": "line_width * 3" },
+        "top_bottom_thickness": {"value":  "layer_height * 6"},
+        "infill_sparse_density": {"default_value": 25},
+        "infill_pattern": {"value": "grid"},
```
Comment (konskarm) [inline resources/definitions/trimaker_cosmosII.def.json:169]:
```suggestion
        "infill_pattern": {"value": "'grid'"},
```

**Review by konskarm — CHANGES_REQUESTED** [review]
Minor fixes. Grid has to be in single quotes as it is a value and not a setting name, or else Cura is searching for a setting called grid.

**Comment by tomasbaldi** [comment]
Thank you very much @konskarm ! I really appreciate the work they do with CURA. I have made the changes you mentioned to me. The COSMOS printing area is offset from the center of the platform. The NEBULA's print area was wrong within the code. Thank you very much again!

**Comment by tomasbaldi** [comment]
Hello! Are the changes I made correct?

**Comment by konskarm** [comment]
Hi @tomasbaldi. Sorry for the long delay, this PR was lost among others. Everything seems to be working as expected now. The only issue I see is the mix of tabs and spaces in the .def.json files, but I will just fix it locally, no need to go back and forth. So I am merging this PR. The Trimaker printers will become available in the next Cura release.

**Pull request merged**